### PR TITLE
Document uniqueness of `name` on `Organization` and `User`

### PR DESCRIPTION
**Pull request by nicksnyder**
Document the uniqueness of the `name` field on the field itself.

Closes https://github.com/bufbuild/registry-proto/issues/2

### CODE REVIEW

**Comment by bufdev** [comment]
I could go either way on the meat of this - moving the unique fields above the `create_time` and `update_time`. The thinking was that the "standard resource fields", namely `id, create_time, update_time`, all come first, where the resource-specific fields come later.

If we did do this, you'd need to update the comments globally, not just for one package (ie all of the `module` package needs to be updated as well). However, I do think having the uniqueness constraints at the top level on a message is nice, so we could perhaps have them both on the field and at the top level.

**Comment by nicksnyder** [comment]
> The thinking was that the "standard resource fields", namely id, create_time, update_time, all come first, where the resource-specific fields come later.

Yeah, I figured.

> If we did do this, you'd need to update the comments globally, not just for one package (ie all of the module package needs to be updated as well). 

Not sure what you mean here. It isn't clear to me that we would want to re-order `name` field in `branch`, `module`, `tag` because those are only identifiers within some context (user/org/module), and not really equivalent to an `id` or `name` in the same sense that it is for `User` and `Organization`.

In any case, after some reflection, I think adding docs to the field is sufficient (which is what this PR diff now is), and we don't need to go so far as changing the order of or renaming the fields.

**Comment by bufdev** [comment]
> Not sure what you mean here.

Yea looking into it, we actually do have "uniqueness" comments on the `module` package types, I wasn't paying attention closely enough. This LGTM.

File: buf/registry/owner/v1beta1/organization.proto [hunk 1]
```diff
@@ -36,6 +36,7 @@ message Organization {
   // The last time the Organization was updated.
   google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
   // The name of the Organization.
+  // A name uniquely identifies an Organization, however name is mutable.
```
Comment (bufdev) [inline buf/registry/owner/v1beta1/organization.proto:39]: Can you put a newline between 38 and 38, with a comment? Same below. So:

```
// The name of the Organization.
//
// A name uniquely identifies an Organization, however name is mutable.
```

Comment (nicksnyder) [reply buf/registry/owner/v1beta1/organization.proto:39]: Yep

```suggestion
  //
  // A name uniquely identifies an Organization, however name is mutable.
```

File: buf/registry/owner/v1beta1/user.proto [hunk 2]
```diff
@@ -36,6 +36,7 @@ message User {
   // The last time the User was updated.
   google.protobuf.Timestamp update_time = 3 [(buf.validate.field).required = true];
   // The name of the User.
+  // A name uniquely identifies a User, however name is mutable.
```
Comment (nicksnyder) [inline buf/registry/owner/v1beta1/user.proto:39]:
```suggestion
  //
  // A name uniquely identifies a User, however name is mutable.
```

**Review by bufdev — APPROVED** [review]

**Pull request merged**